### PR TITLE
feat(nodes): memory item storage

### PR DESCRIPTION
**Pull request by psychedelicious**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [x] Yes
- [ ] No

## Description

Prep for possible upcoming change to not store sessions in the DB.

- Remove unused `get_raw` method from `ItemStorageABC`
- Add `delete` method to `ItemStorageABC` (it was already on the SQLite impl)
- Add memory impl for `ItemStorageABC`

## QA Instructions, Screenshots, Recordings

The app should still run just fine.

## Merge Plan

Do not merge. This requires discussion w/ Invoke team.

### CODE REVIEW

**Review by hipsterusername — APPROVED** [review]

**Comment by psychedelicious** [comment]
convert to draft to prevent accidental merge

File: invokeai/app/services/item_storage/item_storage_memory.py
Comment (maryhipp) [inline invokeai/app/services/item_storage/item_storage_memory.py:37]: I feel like if we aren't using these, lets get rid of the implementation until/unless its needed

File: invokeai/app/services/item_storage/item_storage_memory.py
Comment (maryhipp) [inline invokeai/app/services/item_storage/item_storage_memory.py:11]: Any reason not to go ahead and swap this out in the dependencies?

File: invokeai/app/services/item_storage/item_storage_memory.py
Comment (psychedelicious) [reply invokeai/app/services/item_storage/item_storage_memory.py:37]: Yeah good call

File: invokeai/app/services/item_storage/item_storage_memory.py
Comment (psychedelicious) [reply invokeai/app/services/item_storage/item_storage_memory.py:11]: I suppose not - we'll hold off on merging this til we're ready anyways

**Comment by psychedelicious** [comment]
I've also removed the `list` and `search` methods from the item storage service, and the tests.

Once this merges, we should cut an RC and test it. If all is good (I'm sure it will be), we can add a db migration to drop the `graph_executions` table! That will free up a very solid chunk of disk space for most users.

**Comment by hipsterusername** [comment]
@psychedelicious - Ready for me to hit the button?

**Comment by psychedelicious** [comment]
Needs code review

File: invokeai/app/services/item_storage/item_storage_memory.py
Comment (maryhipp) [inline invokeai/app/services/item_storage/item_storage_memory.py:19]: Is there any sort of cleanup at play here? Won't the dict in memory get really big until the user restarts their server?

Comment (psychedelicious) [reply invokeai/app/services/item_storage/item_storage_memory.py:35]: Whoops! I had a note for myself to do this but forgot. I've implemented `max_size` in two ways:

- With an unordered dict, no guarantees about the order of items: [e49cd87](https://github.com/invoke-ai/InvokeAI/pull/5603/commits/e49cd874879a5eb182e3cb1e9a6e57ec23d2f7c0)
- With an `OrderedDict`, using an LRU eviction strategy: [1d7bc01](https://github.com/invoke-ai/InvokeAI/pull/5603/commits/1d7bc01d138b3b47f02fba09c3b2d89a438cc07a)

I think the latter is the better option.

I also added tests for this both implementations.

File: invokeai/app/services/shared/sqlite_migrator/migrations/migration_5.py
Comment (maryhipp) [inline invokeai/app/services/shared/sqlite_migrator/migrations/migration_5.py:49]: 😮

File: invokeai/app/services/item_storage/item_storage_sqlite.py
Comment (RyanJDick) [inline invokeai/app/services/item_storage/item_storage_sqlite.py:12]: Can we delete this class altogether?

Seems like it's unused at this point. And, I think it might be the last remaining piece of code that changes the SQL schema outside of the migrator, so I don't want anyone to be tempted to revive this as-is.

**Review by RyanJDick — APPROVED** [review]
LGTM, just left one question

**Pull request merged**
